woocommerce/woocommerce-android
# Display quantity rules in product variations

⚠️  This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2699)

Closes: #8400
Closes: #8401

### Why
We are adding read-only support to the Min/Max Quantities extension

### Description
This PR adds the necessary changes to display the product variation's quantity rules on the variations details screen. It does that by adding the new `quantityRules` extension function to the `VariationDetailCardBuilder`. The `GetProductVariationQuantityRules ` is used to get the quantity rules depending on the availability of the Min / Max extension. In this first iteration, the Quantity Rules card is not shown if the quantity rules are empty, we might want to change this behavior in the future when adding editing support. 

### Testing instructions
TC1
1. Open the Products tab
2. Select any variable product containing a variation with quantity rules
3. Tap on Variations
4. Tap on the variation with the quantity rules
5. Check that the Quantity Rules card is shown
6. Tap on the Quantity Rules card
7. Check that the app navigates to the Quantity Rules fragment
8. Press back
9. Check that the app returns to the variation details screen

TC2
1. Complete the steps above with a variation without quantity rules and check that the Quantity Rules card is NOT shown

### Images/gif

https://user-images.githubusercontent.com/18119390/229588623-4fb19847-d1d2-4662-baff-cde7c63139e2.mp4
